jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -293,7 +293,9 @@ static const uint8_t ecma_uint4_clz[] = { 4, 3, 2, 2, 1, 1, 1, 1, 0, 0, 0, 0, 0,
 static ecma_number_t
 ecma_utf8_string_to_number_by_radix (const lit_utf8_byte_t *str_p, /**< utf-8 string */
                                      const lit_utf8_byte_t *end_p, /**< end of utf-8 string  */
-                                     uint32_t radix) /**< radix */
+                                     uint32_t radix, /**< radix */
+                                     bool allow_underscore)  /**< true - if '_' characters are allowed in the string
+                                                                  false - otherwise */
 {
   JERRY_ASSERT (radix == 2 || radix == 8 || radix == 16);
```

Comment:
Since radix is very limited, maybe we could use a flag bit to represent `allow_underscore`.